multinet/db.py

```diff
@@ -222,13 +229,38 @@ def get_workspace_metadata(name: str) -> Workspace:
 
     # Find the metadata record for the named workspace. If it's not there,
     # something went very wrong, so bail out.
-    metadata = workspace_mapping(name)
+    metadata = workspace_mapping.__wrapped__(name)
     if metadata is None:
         raise DatabaseCorrupted()
 
     return metadata
 
 
+def set_workspace_permissions(
+    name: str, permissions: WorkspacePermissions
+) -> WorkspacePermissions:
+    """Update the permissions for a given workspace."""
+    if not workspace_exists(name):
+        raise WorkspaceNotFound(name)
+
+    doc = copy.deepcopy(workspace_mapping(name))
+    if doc is None:
+        raise DatabaseCorrupted()
+
+    # TODO: Do user object validation once ORM is implemented
+
+    # To ensure owner is only changed explicitly
```

Comment:
What do you mean by this?

Reply:
I mean that the owner can't be changed through setting the workspace permissions like everything else, it needs to be explicitly changed through it's own endpoint or something.

Reply:
Oh I see. I'd suggest different wording for the comment:

```suggestion
    # Disallow changing workspace ownership through this function.
```